internal/socket/socket.go

```diff
@@ -0,0 +1,174 @@
+//go:build windows
+
+package socket
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"syscall"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"golang.org/x/sys/windows"
+)
+
+//go:generate go run golang.org/x/sys/windows/mkwinsyscall -output zsyscall_windows.go socket.go
+
+//sys getsockname(s windows.Handle, name unsafe.Pointer, namelen *int32) (err error) [failretval==socketError] = ws2_32.getsockname
+//sys getpeername(s windows.Handle, name unsafe.Pointer, namelen *int32) (err error) [failretval==socketError] = ws2_32.getpeername
+//sys bind(s windows.Handle, name unsafe.Pointer, namelen int32) (err error) [failretval==socketError] = ws2_32.bind
+
+const socketError = uintptr(^uint32(0))
+
+var ErrSocketClosed = fmt.Errorf("socket closed: %w", net.ErrClosed)
+
+// CloseWriter is a connection that can disable writing to itself.
+type CloseWriter interface {
```

Comment:
Are these used/returned from anywhere here?

Reply:
This and CloseReader

Reply:
`ErrSocketClosed` is returned hvsock.go, CloseReader/Writer are leftovers, ill remove

Reply:
Idk why it highlighted this whole block.. just meant CloseWriter and CloseReader 😐 but okay sweet! Resolving